PCbuild/pythoncore.vcxproj

```
@@ -716,6 +716,26 @@
     <Delete Files="$(IntDir)pyconfig.h;$(OutDir)pyconfig.h" />
   </Target>
 
+  <Target Name="EnsureClangProfileDir" BeforeTargets="PrepareForBuild"
+          Condition="$(PlatformToolset) == 'ClangCL' and $(Configuration) == 'PGInstrument' and !Exists($(ClangProfileDir))"> 
```

```suggestion
          Condition="$(PlatformToolset) == 'ClangCL' and $(Configuration) == 'PGInstrument'"> 
```

Target conditions are best to be very static, based on input arguments, rather than external state (at least at this level - when using them to process items it's often different).

Making a directory that exists should be a safe no-op.

---

Great! Will try. In case it does not work out, I can move the condition into
`<MakeDir Directories="$(ClangProfileDir)" !Exists($(ClangProfileDir)) />`
I had it there in the beginning.